scripts/globals/pets/Kupipi.lua

```diff
@@ -0,0 +1,283 @@
+	mob:setLocalVar("flashTime",0)
+
+	mob:addListener("TRUST_COMBAT_TICK", "KUPIPI_BUFF_TICK" .. kupipi, function(mob, player, target)
+	    local battletime = os.time()
+		local buffTime = mob:getLocalVar("buffTime")
+
+		if (battletime > buffTime + buffCooldown) then
+		    doBuff(mob, player)
+		end
+	end)
+
+	mob:addListener("TRUST_COMBAT_TICK", "KUPIPI_AILMENT_TICK" .. kupipi, function(mob, player, target)
+	    local battletime = os.time()
+		local ailmentTime = mob:getLocalVar("ailmentTime")
+
+		if (battletime > ailmentTime + ailmentCooldown) then
+		    local spell = doStatusRemoval(mob, player)
+			if (spell > 0 ) then
+		        mob:castSpell(spell, player)
+			end
+			mob:setLocalVar("ailmentTime",battletime)
+		end
+	end)
+
+	mob:addListener("TRUST_COMBAT_TICK", "KUPIPI_DEBUFF_TICK" .. kupipi, function(mob, player, target)
+	    local battletime = os.time()
+		local debuffTime = mob:getLocalVar("debuffTime")
+
+		if (battletime > debuffTime + debuffCooldown) then
+		    local spell = doDebuff(mob, target)
+			if (spell > 0 ) then
+		        mob:castSpell(spell, target)
+			end
+			mob:setLocalVar("debuffTime",battletime)
+		end
+	end)
+
+	mob:addListener("TRUST_COMBAT_TICK", "KUPIPI_CURE_TICK" .. kupipi, function(mob, player, target)
+	    local battletime = os.time()
+		local cureTime = mob:getLocalVar("cureTime")
+
+		if (battletime > cureTime + cureCooldown) then
+		    local party = player:getParty()
+            for _,member in ipairs(party) do
+                if (member:getHPP() <= 30) then
+					local spell = doEmergencyCure(mob)
+					if (spell > 0) then
+                        mob:castSpell(spell, member)
+					    mob:setLocalVar("cureTime",battletime)
+                        break
+					end
+                elseif (member:getHPP() <= 75) then
+					local spell = doCure(mob)
+					if (spell > 0) then
+                        mob:castSpell(spell, member)
+					    mob:setLocalVar("cureTime",battletime)
+                        break
+					end
+			    end
+			end
+			mob:setLocalVar("cureTime",battletime - 4)  -- If no member has low HP change global check to 8 seconds
+		end
+	end)
+
+	mob:addListener("TRUST_COMBAT_TICK", "KUPIPI_COMBAT_TICK" .. kupipi, function(mob, target)
+	    if (mob:getTP() > 1000) then
+		    weaponskill = doKupipiWeaponskill(mob)
+			mob:useMobAbility(weaponskill, mob)
+		end
+	end)
+end
+
+function doStatusRemoval(mob, player)
+	local mp = mob:getMP()
+	local lvl = mob:getMainLvl()
+	local spell = 0
+
+	if (player:hasStatusEffect(dsp.effect.POISON) and lvl >= 6 and mp >= 8) then
+	    spell = 14
+	elseif (player:hasStatusEffect(dsp.effect.PARALYSIS) and lvl >= 9 and mp >= 12) then
+	    spell = 15
+	elseif (player:hasStatusEffect(dsp.effect.BLINDNESS) and lvl >= 14 and mp >= 16) then
+	    spell = 16
+	elseif (player:hasStatusEffect(dsp.effect.SILENCE) and lvl >= 19 and mp >= 24) then
+	    spell = 17
+	elseif (player:hasStatusEffect(dsp.effect.CURSE_I) and lvl >= 29 and mp >= 30) then
```

Comment:
Kupipi also casts cursna to attempt to remove doom and if I remember correctly (not 100% sure though) she prioritizes it over other status effects.